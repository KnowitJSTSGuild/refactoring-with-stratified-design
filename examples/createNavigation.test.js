// @vitest-environment jsdom
import { describe, it, expect } from "vitest";
import { createNavigation } from "./createNavigation";
import { screen } from "@testing-library/dom";

describe("createNavigation", () => {
  it("Prints a navigation list with subitems", () => {
    const items = [
      { name: "First item", type: "text" },
      { name: "Second item", type: "text" },
      {
        name: "A nested one",
        type: "section",
        items: [{ name: "First subitem", type: "text" }],
      },
    ];
    createNavigation(items);
    expect(document.body).toMatchSnapshot();
  });

  it("Prints Untitled for items without names", () => {
    createNavigation([{ name: "", type: "text" }]);
    expect(screen.getByText("Untitled")).not.toBeFalsy();
  });
});
