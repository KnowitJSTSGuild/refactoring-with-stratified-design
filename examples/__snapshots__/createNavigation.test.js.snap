// Vitest Snapshot v1

exports[`createNavigation > Prints a navigation list with subitems 1`] = `
<body
  style="overflow: auto;"
>
  <div>
    <h3
      class="hlpresentation"
      id="defaultcontentheader"
    >
      Sisältö
    </h3>
    <ul
      class="hlnavsection"
      id="navigator_sectionlist"
    >
      <li
        class="unhlsection"
      >
        <button>
          First item
        </button>
      </li>
      <li
        class="unhlsection"
      >
        <button>
          Second item
        </button>
      </li>
      <li
        class="unhlsection"
      >
        <button>
          A nested one
        </button>
        <ul>
          <li
            class="unhlsection"
          >
            <button>
              First subitem
            </button>
          </li>
        </ul>
      </li>
    </ul>
  </div>
  <section
    class=""
    id="navigatorcontainer"
  >
    <section
      class="navchildsec"
      id="section_nav"
    />
    <section
      class="navchildsec"
      id="previewer"
    />
    <div
      class=""
      id="addedcontentparent"
    >
      <h3
        class="unhlpresentation"
        id="addedcontentheader"
      >
        Lisätty sisältö
      </h3>
      <div
        class=""
        id="addedcontent"
      />
    </div>
  </section>
</body>
`;
